### ISSUE TITLE
Mixer needs a pan knob

### ISSUE THREAD

**Issue by Reaper10** [LMMS/lmms#1974]
All mixer strips needs an pan knob on them.

**Comment by Umcaruje** [comment]
I think a Pan knob would just overcrowd the mixer. You can pan any instrument already, and if you need panning on a mixer track, you can simply add an amplifier and pan with it: 
![image](https://cloud.githubusercontent.com/assets/6282045/7215644/dbcfbe2e-e5e1-11e4-94cd-01bee9c66604.png)

**Comment by unfa** [comment]
That's true, although a bit strange, looking at other DAWs.

**Comment by tresf** [comment]
Agreed with @Umcaruje.  We simply don't have room for this in our current interface.  Closing as wontfix.

**Comment by Reaper10** [comment]
Ardour  has one in the mixer

**Comment by Wallacoloo** [comment]
@Reaper10 Cool. But as you probably noticed by now, this is not the Ardour forum.

**Comment by Wallacoloo** [comment]
@Reaper10 The only way in which bringing Ardour (or any other DAW) into the conversation could be useful is if you show that Ardour had a similar problem with overcrowding their mixer after adding a pan knob and then describe how they solved it, and explain how their experience might be useful to lmms.

**Comment by Reaper10** [comment]
When the sends changes from this 
![8aed812c-c281-11e4-9093-3d0599010c32 1](https://cloud.githubusercontent.com/assets/8137561/8294347/81c7b320-1901-11e5-9b1c-9339c1212917.png) to this 
![e9366c66-c282-11e4-9fca-535c462495cc 1](https://cloud.githubusercontent.com/assets/8137561/8294367/c6d85258-1901-11e5-8afe-ec5384d6e48d.png) you can put pan knob where the send knob is right now.

**Comment by Umcaruje** [comment]
Nobody said sending will change to that. Imo that way of routing seems way clunky and bulky for the UI.

**Comment by Reaper10** [comment]
You could just move the send from top of the mixer to bottom of the mixer, then add the pan where the send was.

**Comment by miketurn** [comment]
I am aware that this request has been closed but I thought I would share some opinions.
(Not to say that it is right or wrong, but to be honest I believe that LMMS is the only mixer I have ever seen that does not have pan knobs.)

I admit I have been on the fence with this one quite a bit as well, I do love the simplistic look of the mixer but on the other hand pan knobs could be helpful.

I am aware that you can use the pan knobs for individual tracks or you can add an effect to your mixer channel that contains panning, but these methods can feel a little odd at times.
If you are in the mixer, mixing a song it can be a little distracting to have to break focus from the mixer and go find the specific track to pan something, or to have to open an effects "controls" to see if the item has been panned or to adjust its settings.
Having pan knobs in the mixer allows you to see a visual right away for each specific mixer channel.

Again this all depends on the user and how much they use panning?
Like I said I agree that it is a tough decision, but may be worth taking a more indepth look at and considering a possible solution for.

I will link my post about adding the ability to "vertically stretch the mixer" here which can possibly give the room for pan knobs if desired.
I always thought it might be a nice idea to give certain items in the mixer a "modular" ability, allowing you to show/hide elements that you want or don't want.
Anyways that's about it, if anyone has any opinions please share them, good or bad.

**Comment by tresf** [comment]
One way to keep the real-estate and still fulfill the original request is to offer a hidden L/R slider with a small graphic on the knob to illustrate fade direction.

![screen shot 2017-01-26 at 8 12 40 pm](https://cloud.githubusercontent.com/assets/6345473/22358830/67926d34-e410-11e6-984c-f9854231d149.png)

**Comment by AndiEcker** [comment]
@tresf great idea / proposal

2017-01-27 2:43 GMT+00:00 Tres Finocchiaro <notifications@github.com>:

**Comment by Spekular** [comment]
@tresf how does one ctrl+drag the pan slider to automate it?

I could see the handle+drag being pan, and an empty part of the fader+drag being volume, but that seems a bit clunky.

**Comment by miketurn** [comment]
@tresf
Hello, it has been a while.
I forgot that my comment generated a couple of responses here, good to see that others are possibly still interested in this feature.  Hopefully this issue gets "Opened" again someday, rare to see an idea on this site completely closed the way this one was.
I have been trying to convince someone I know to join this forum, most of my ideas have actually come from him directly and or have been part of conversations we have had about LMMS.  Panning in the mixer was a desired feature of his as well, and I know he has some ideas.  He actually has provided most of the graphic examples from my posts so I know he can provide some visuals of maybe how this feature can be fit into the mixer.
If I can get him to join I will mention to post some ideas of his here on this thread, if not I will ask him about them and maybe I can share them myself.

**Comment by BaraMGB** [comment]
I think this is a valid issue. We should reopen this. @tresf?

**Comment by miketurn** [comment]
@tresf 
@BaraMGB 
Thank You for this

**Comment by Hussam-Eddin-Alhomsi** [comment]
I think that mixer channels need a pan parameter for consistency and I think that controlling the pan parameter with a knob is the best choice.
Instrument plugins and sample tracks have volume and pan parameters, why doesn't the mixer have both of them too?

@Umcaruje 

> I think a Pan knob would just overcrowd the mixer. You can pan any instrument already, and if you need panning on a mixer track, you can simply add an amplifier and pan with it.

Then there is no need for the volume slider, too, right?!

@tresf 

> without using up any more space with the GUI.

I think that the problem that's keeping us from adding this feature comes from the fixed height of the mixer, I think that this should be changed.
I don't know how complex it is to increase the height of the mixer or make the mixer's height vector-based, but a mixer with more height will allow finer volume sliders adjusting and more precise levels metering.

**Comment by tresf** [comment]
> but a mixer with more height will allow finer volume sliders adjusting and more precise levels metering.

 * Sometimes too much granularity is just obnoxious.  Finer control can be achieved by a shortcut key without making a monster mixer. 
 * Just because the mixer CAN be taller, that doesn't necessarily mean it should be by default.
 * Making the sliders taller is really a separate enhancement.  Unless you have a new UI proposal, it's nothing to do with the pan as pan wouldn't occupy the same real estate, so the height of the volume is really not related to this request.  A vertically resizable mixer is a separate topic.

Adding a PAN knob is just a bunch of unused real estate.  I would defer this decision to the @budislav's and @RebeccaLaVie's of the world.  Perhaps [as part of the single window design](https://github.com/LMMS/lmms/issues/1911), but something we can roll out sooner.

![image](https://user-images.githubusercontent.com/6345473/27544995-ae2990c0-5a5c-11e7-83d0-a7439e2784b2.png)

**Comment by tresf** [comment]
I'd also like to mention the 80% / 20% rule, which lends itself to @Umcaruje's point in the second post.  The UI should show the most commonly used components.  Although I don't think it should require a separate plugin to pan, it's not done as commonly as the volume, so the UI should reflect that.

**Comment by Umcaruje** [comment]
there is less vertical real estate than there is horizontal, if you take a standard 4:3 or 16:9. I think the mixer should not get any higher. The way a pan knob would be implemented would be to actually make the mixer stripes wider and seperate the LED's from the volume fader. I'm going to post a screenshot of bitwig cause I don't have time to make a mockup rn 
![image](https://user-images.githubusercontent.com/6282045/27550121-c897b940-5a9e-11e7-98f0-b9d1d8128fb1.png) 
Though the LED's should be wider, but the design would be a step into that direction

**Comment by BaraMGB** [comment]
Don't let us make this one more complicated. A knob has how many pixels in height? 23? That is not a problem since the mixer is a floating window. The trend goes to bigger displays with higher solutions. No one makes music with a Netbook.

**Comment by RebeccaLaVie** [comment]
Fitting the pan knob into the fx mixer and extending the height vertically might work as a quick fix, but I don't think that will be a good long-term solution. I would have to agree with @Umcaruje about being mindful of our vertical space and with @tresf about keeping a clear hairarchy by not overcrowding the elements that are most important to the user.

![27282273-86b77fd8-54ef-11e7-84d2-6c0ec5a571c3](https://user-images.githubusercontent.com/7960169/27552840-a46f6702-5a5d-11e7-98d7-e1fcc55c9ab8.png)

Maybe we could incorporate some of @budislav's work into this? It would also bring the UI one step closer to his proposal.

**Comment by budislav** [comment]
#1911 design updated. Pan is needed in mixer from the very beginning.

**Comment by RebeccaLaVie** [comment]
Hmm, I might have attached the wrong screenshot. @budislav I am proposing that we try to match your layout for the FX mixer with the pan knob.

**Comment by budislav** [comment]
@RebeccaLaVie my mistake :) I didn't said that I updated design at that moment. Your first image was right :)

**Comment by Sawuare** [comment]
How about a horizontal slider for panning?I think it will look great with the vertical slider for volume.
Whatever the widget for the panning is, I think it should be positioned above the volume slider.

The design I'm describing can be seen in the mixer of Qtractor.

**Comment by tresf** [comment]
As part of a pruning effort, this `enhancement` request is archived into a dedicated "Better Workflow" checklist here #4877.
